src/Nethermind/Nethermind.Synchronization/FastSync/PendingSyncItems.cs

```diff
@@ -168,8 +168,26 @@ public List<StateSyncItem> TakeBatch(int maxSize)
             // the moment we find the first leaf we will know something more about the tree structure and hence
             // prevent lot of Stream2 entries to stay in memory for a long time 
             int length = MaxStateLevel == 64 ? maxSize : Math.Max(1, (int)(maxSize * ((decimal)MaxStateLevel / 64) * ((decimal)MaxStateLevel / 64)));
-
             List<StateSyncItem> requestItems = new(length);
+
+            if (CodeItems.Count > 0)
+            {
+                length = Math.Min(length, CodeItems.Count);
+
+                for (int i = 0; i < length; i++)
+                {
+                    if(CodeItems.TryPop(out var codeItem))
+                    {
+                        requestItems.Add(codeItem!);
+                    }
+                }
+
+                if(requestItems.Count > 0)
```

Comment:
can there be a case where this does not happen?
if yes, then the value of length should also be modified to the previous value?

Reply:
Agree, we should assign it to a new variable.